source/client-side-encryption/tests/unified/createKey-dataKeyOpts-keyMaterial.yml

```diff
@@ -0,0 +1,65 @@
+description: createKey-dataKeyOpts-keyMaterial
+
+schemaVersion: "1.8"
+
+runOnRequirements:
+  - csfle: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents:
+        - commandStartedEvent
+        - commandSucceededEvent
+        - commandFailedEvent
+  - clientEncryption:
+      id: &clientEncryption0 clientEncryption0
+      clientEncryptionOpts:
+        keyVaultClient: *client0
+        keyVaultNamespace: keyvault.datakeys
+        kmsProviders:
+          local:
+            key: { $$placeholder: 1 }
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &keyvault keyvault
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &datakeys datakeys
+
+initialData:
+  - databaseName: *keyvault
+    collectionName: *datakeys
+    documents: []
+
+tests:
+  - description: create datakey with custom key material
+    operations:
+      - name: createKey
+        object: *clientEncryption0
+        arguments:
+          kmsProvider: local
+          opts:
+            # "key_material" repeated 8 times.
+            keyMaterial: { $binary: { base64: "a2V5X21hdGVyaWFsa2V5X21hdGVyaWFsa2V5X21hdGVyaWFsa2V5X21hdGVyaWFsa2V5X21hdGVyaWFsa2V5X21hdGVyaWFsa2V5X21hdGVyaWFsa2V5X21hdGVyaWFs", subType: "00" } }
+        expectResult:
+          id: { $$type: binData }
```

Comment:
`createKey` is documented as returning a Binary object. Where does the `id` field come from here?

Should this test instead have `expectResult: { $type: binData }`?

Reply:
This is an inconsistency introduced during rebasing of other Key Management API drafts. I had wording in the unified test format spec that instructed drivers to report the value of the Binary object as `{ id: (value) }` because I was under the impression that `expectResult: { $$type: binData }` would not work as expected. However, it seems that it _does_ work as suggested, so I will go with that instead.